kong/db/schema/typedefs.lua

```diff
@@ -521,6 +521,23 @@ local function validate_path_with_regexes(path)
   return true
 end
 
+local function validate_regex_or_plain_pattern(pattern)
```

Comment:
This code is very similar to: https://github.com/Kong/kong/blob/master/kong/db/schema/typedefs.lua#L492

Do you think we could adjust parts of these two to keep the logic of validating if something is a valid prefix or regex?